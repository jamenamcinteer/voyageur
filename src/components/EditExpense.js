import React from "react";
import TripHeader from "./Navigation/TripHeader";
import { connect } from "react-redux";

import ExpenseForm from "./Forms/ExpenseForm";

const EditExpense = props => {
  const trip = props.trips.find(trip => trip._id === props.match.params.id);

  const expense = props.expenses.find(
    i => i._id === props.match.params.expenseId
  );

  return (
    <div>
      {expense && (
        <div>
          <TripHeader
            title="Edit Expense"
            theme={props.theme}
            backTo={`/trip/${props.match.params.id}/budget-category/${
              expense.budgetCategoryId
            }/budget-item/${expense.budgetItemId}`}
            trip={trip}
            budgetItems={props.budgetItems}
            expenses={props.expenses}
            auth={props.auth}
          />
          <ExpenseForm
            expense={expense}
            theme={props.theme}
            trip={trip}
            budgetCategories={props.budgetCategories}
            budgetItems={props.budgetItems}
            history={props.history}
          />
        </div>
      )}
    </div>
  );
};

const mapStateToProps = (state, props) => {
  return {
    trips: state.trips,
    budgetItems: state.budgetItems,
    budgetCategories: state.budgetCategories,
    expenses: state.expenses,
    auth: state.auth
  };
};

export default connect(mapStateToProps)(EditExpense);
