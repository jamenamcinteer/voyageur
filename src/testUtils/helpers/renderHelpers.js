import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import theme from "../../theme";
import "jest-styled-components";
import { createStore } from "redux";
import { Provider } from "react-redux";
import tripsReducer, { tripsReducerDefaultState } from "../../reducers/trips";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";

export function renderWithTheme(component) {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
}

export function renderWithReduxRouterAndTheme(
  ui,
  {
    initialState,
    store = createStore(tripsReducer, tripsReducerDefaultState),
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) {
  return {
    ...render(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <Router history={history}>{ui}</Router>
        </ThemeProvider>
      </Provider>
    ),
    store,
    history
  };
}
