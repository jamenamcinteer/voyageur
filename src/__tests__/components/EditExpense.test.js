import React from "react";
import { cleanup } from "@testing-library/react";
import { createStore } from "redux";
import "react-dates/initialize";
import EditExpense from "../../components/EditExpense";
import {
  trips,
  budgetCategories,
  budgetItems,
  expenses,
  auth
} from "../../testUtils/fixtures/mockStoreData";
import { renderWithReduxRouterAndTheme } from "../../testUtils/helpers/renderHelpers";

jest.mock("uniqid", () => {
  let value = 1;
  return () => value++;
});

afterEach(cleanup);

test("can render with redux with custom store", () => {
  const store = createStore(() => ({
    trips,
    budgetCategories,
    budgetItems,
    expenses,
    auth
  }));
  const { asFragment } = renderWithReduxRouterAndTheme(
    <EditExpense match={{ params: { id: "1", expenseId: "1" } }} />,
    {
      store
    }
  );
  expect(asFragment()).toMatchSnapshot();
});
