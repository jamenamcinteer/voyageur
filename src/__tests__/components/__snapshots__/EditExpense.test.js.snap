// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render with redux with custom store 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div>
        <div>
          <div
            class="sc-bdVaJa hZqfbl"
          >
            <a
              href="/trip/1/budget-category/1/budget-item/1"
            >
              <svg
                aria-label="Go Back"
                style="fill: #4b5c6b; width: 20px; height: 20px; float: left; margin-top: 2px;"
                version="1.1"
                viewBox="0 0 199.404 199.404"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <polygon
                    points="135.412,0 35.709,99.702 135.412,199.404 163.695,171.119 92.277,99.702 163.695,28.285 	"
                  />
                </g>
              </svg>
            </a>
            <h1
              class="sc-bwzfXH OVfcq"
            >
              Edit Expense
            </h1>
            <button
              class="sc-htpNat fqCEYg"
            >
              <img
                alt="Account Options"
                class="sc-bxivhb jwtarI"
                src="https://picsum.photos/id/947/45/45"
              />
            </button>
          </div>
          <div
            style="position: relative;"
          >
            <div
              style="background-image: url(https://images.unsplash.com/photo-1549317487-92cec5e76645?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjc5ODc0fQ&w=300&h=100&fit=crop&crop=focalpoint); background-repeat: no-repeat; background-size: cover; filter: brightness(0.6); height: 150px;"
            />
            <div
              style="height: 150px; padding: 0px 20px; color: rgb(255, 255, 255); font-size: 1em; font-weight: normal; position: absolute; width: 100%; top: 0px;"
            >
              <div
                style="display: flex; justify-content: space-between; align-items: center; height: 90px;"
              >
                <h2>
                  Hawaii
                </h2>
                <h3>
                  Apr 26, 1970 - Oct 1, 8307
                </h3>
              </div>
              <div
                style="width: 100%; height: 25px; position: relative; background: rgb(220, 224, 227); border-radius: 5px;"
              >
                <span
                  style="display: block; height: 100%; border-radius: 5px; position: relative; overflow: hidden; width: 29.731%; background-color: rgb(12, 124, 136);"
                />
                <div
                  style="position: absolute; top: 0px; color: white; left: 10px; line-height: 25px;"
                >
                  $595 of $2000
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-kgoBCf RJSBM"
      >
        <div
          class="sc-dnqmqq jnbCEU"
        >
          <label
            class="sc-iwsKbI dtSDPi"
            for="7"
          >
            Budget Category
          </label>
          <select
            class="sc-gZMcBi byovWb"
            id="7"
          >
            <option
              value=""
            >
              Select category...
            </option>
            <option
              value="1"
            >
              Airfare
            </option>
          </select>
        </div>
        <div />
        <div
          class="sc-dnqmqq jnbCEU"
        >
          <label
            class="sc-iwsKbI dtSDPi"
            for="8"
          >
            Budget Item
          </label>
          <select
            class="sc-gZMcBi byovWb"
            id="8"
          >
            <option
              value=""
            >
              Select budget item...
            </option>
            <option
              value="1"
            >
              Flights
            </option>
          </select>
        </div>
        <div />
        <div
          class="sc-bZQynM blxRiI"
        >
          <label
            class="sc-gzVnrw sohPS"
            for="9"
          >
            Summary of Expense
          </label>
          <input
            class="sc-htoDjs hNiZih"
            id="9"
            type="text"
            value="Southwest"
          />
        </div>
        <div />
        <div
          class="sc-dnqmqq jnbCEU"
        >
          <label
            class="sc-iwsKbI dtSDPi"
            for="10"
          >
            Currency
          </label>
          <select
            class="sc-gZMcBi byovWb"
            id="10"
          >
            <option
              value="USD"
            >
              USD ($)
            </option>
          </select>
        </div>
        <div />
        <div
          class="sc-bZQynM blxRiI"
        >
          <label
            class="sc-gzVnrw sohPS"
            for="11"
          >
            Amount Spent ($)
          </label>
          <input
            class="sc-htoDjs hNiZih"
            id="11"
            placeholder="0.00"
            type="text"
            value="594.62"
          />
        </div>
        <div />
        <h4
          class="sc-jKJlTe gyzmAq"
        >
          Date of Expense
        </h4>
        <div
          class="SingleDatePicker SingleDatePicker_1"
        >
          <div>
            <div
              class="SingleDatePickerInput SingleDatePickerInput_1 SingleDatePickerInput__withBorder SingleDatePickerInput__withBorder_2"
            >
              <div
                class="DateInput DateInput_1"
              >
                <input
                  aria-describedby="DateInput__screen-reader-message-date"
                  aria-label="Date"
                  autocomplete="off"
                  class="DateInput_input DateInput_input_1"
                  id="date"
                  name="date"
                  placeholder="Date"
                  type="text"
                  value="04/26/1970"
                />
                <p
                  class="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
                  id="DateInput__screen-reader-message-date"
                >
                  Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
                </p>
              </div>
            </div>
          </div>
        </div>
        <div />
        <div
          class="sc-gqjmRU jgsBPr"
        >
          <label
            class="sc-VigVT hqhiep"
            for="12"
          >
            Notes (optional)
          </label>
          <textarea
            class="sc-jTzLTM cHGIjj"
            id="12"
            type="text"
          />
        </div>
        <div
          class="sc-dxgOiQ VLeUZ"
        >
          <div
            class="sc-kpOJdX ibexlj"
          >
            <a
              backgroundcolor="transparent"
              border="0"
              buttontype="link"
              buttonwidth="auto"
              class="sc-kAzzGY fsudaQ"
              href="/trip/1"
              style="padding: 10px 0px;"
            >
              <span
                class="sc-chPdSV IudMt"
              >
                Cancel
              </span>
            </a>
            <button
              class="sc-jzJRlG hjpZFG"
            >
              <span
                class="sc-cSHVUG hQiayu"
                color="#fff"
              >
                Save
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
